packages/runtime/src/interpreter/interpreter.ts

```diff
@@ -215,31 +215,50 @@ export class Interpreter {
 	): AccountStoreI {
 		let account: AccountStoreI | undefined;
 		let address: string;
-		if (typeof accountRef === "bigint") {
-			if (accountRef === 0n) {
-				address = encodeAddress(this.runtime.ctx.tx.snd);
-				account = this.runtime.ctx.state.accounts.get(address);
-			} else {
-				const accIndex = accountRef - 1n;
-				checkIndexBound(Number(accIndex), this.runtime.ctx.tx.apat ?? [], line);
-				let pkBuffer;
-				if (this.runtime.ctx.tx.apat) {
-					pkBuffer = this.runtime.ctx.tx.apat[Number(accIndex)];
-				} else {
-					throw new Error("pk Buffer not found");
-				}
-				address = encodeAddress(pkBuffer);
-				account = create
-					? this.createAccountIfAbsent(address)
-					: this.runtime.ctx.state.accounts.get(address);
-			}
+
+		if (accountRef === 0n) {
+			address = encodeAddress(this.runtime.ctx.tx.snd);
+			account = this.runtime.ctx.state.accounts.get(address);
+			return this.runtime.assertAccountDefined(address, account, line);
+		} else if (typeof accountRef === "bigint") {
+			return this._getAccountFromReference(accountRef, line, create);
```

Comment:
You are revering the condition? Please check if this is really needed.

Reply:
I am not sure which part are you referring to

Reply:
`typeof accountRef === "bigint"` when this is true we were doing something but we are doing the opposite.

Reply:
we are doing the same it was just moved to a separate method